test-utils/shared/image.tsx

```diff
@@ -0,0 +1,68 @@
+import {render, screen} from '@testing-library/react';
+import React, {ElementType} from 'react';
+
+import {isCompressible} from '../../src/utils/imageCompress';
+
+import {QA} from './types';
+
+export const testSourceProps = <T,>({
+    component: Component,
+    props,
+}: {
+    component: ElementType;
+    props: T &
+        QA & {
+            src?: string;
+            tablet?: string;
+            desktop?: string;
+            mobile?: string;
+            disableCompress?: boolean;
+        };
+}) => {
+    const src = props.src || props.desktop;
+
+    const disableWebp = src && (props.disableCompress || !isCompressible(src));
+
+    render(<Component {...props} />);
+
+    if (!src) {
+        expect(1).toBe(0);
```

Comment:
What does that means?

Reply:
It was the protection when we do not pass `src` or `desktop`.
Actually, test can fall down soft, when they are completed and show what test have not passed.
And test can fall down hard, when test are not executed at all. This can happen, for example, when we try to get element that is not rendered.
In this particular case, not passed `src` or `desktop` means not rendered Component.
But I have learnt how to avoid falling down hard.
`screen.getBy[Role|TestId]` -> `screen.queryBy[Role|TestId]`
I made fix